localstack/services/apigateway/helpers.py

```diff
@@ -658,41 +659,66 @@ def add_path_methods(rel_path: str, parts: List[str], parent_id=""):
             method_resource = create_method_resource(resource, field, field_schema)
             method_resource.request_parameters = method_integration.get("requestParameters")
             responses = field_schema.get("responses", {})
-            for status_code in responses:
+            for status_code, response in responses.items():
                 response_model = None
-                if model_schema := responses.get(status_code, {}).get("schema", {}):
+                if model_schema := response.get(
```

Comment:
This clearly has improved on functionality but I find it hard to read. Let's take this example from swagger documentation.

```
{
  "200": {
    "description": "a pet to be returned",
    "schema": {
      "$ref": "#/definitions/Pet"
    }
  },
  "default": {
    "description": "Unexpected error",
    "schema": {
      "$ref": "#/definitions/ErrorModel"
    }
  }
}
```

The response object keys are **status codes** and the **default** key, so lookups were natural to read before, response-> status code -> schema. Now we destruct early (status_code, response) and that makes a lookup less intuitive and the status code variable is distantly far from where it's used again.

Reply:
I'm gonna look at this better tomorrow, but I was a bit confused by the `responses.get(status_code, {})` part, because the `status_code` would always be in `responses` as its part of its key? I'll look at the overall loop again better.

Reply:
I reworked this part of the importer, I've added more tests now and there were still missing fields and misplaced ones, especially `requestParameters` for `Method`.
I've also commented more the code, explained what we were creating and why, from which part of the imported file.

If you could review this part of the code again, that would be awesome! 🙏 

Also, the example you shared, I don't think we're supporting that at the moment, as we would try to create a `MethodResponse` with a `default` value for `response_code`. We will need to create an AWS validated test with this kind of spec.